flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -186,7 +186,12 @@ object ScalarOperatorGens {
 
       case (TIME_WITHOUT_TIME_ZONE, INTERVAL_DAY_TIME) =>
         generateOperatorIfNotNull(ctx, new TimeType(), left, right) {
-          (l, r) => s"$l $op ((int) ($r))"
+          (l, r) => s"($l $op ((int) ($r))) % ${MILLIS_PER_DAY}"
```

Comment:
also why is the `INTERVAL_DAY_TIME` get cast into `(int)` instead of `(long)` ?

Reply:
could you verify that this cast is safe?

Reply:
> could you verify that this cast is safe?

well, Thank you very much, I have change it.